modules/services/preferences/src/main/java/au/com/shiftyjelly/pocketcasts/preferences/SettingsImpl.kt

```diff
@@ -264,6 +265,33 @@ class SettingsImpl @Inject constructor(
         editor.apply()
     }
 
+    override fun syncOnMeteredNetwork(): Boolean {
```

Comment:
🤔 The more I think about this one, I'm not sure. All those methods depend on the setting `syncOnMeteredNetwork` to derive the proper response. So if they get split into another class, it seems like it might add unnecessary complexity to accessing settings. But I dunno 🤷 

There's other getter methods like `getPodcastsSortType` that have "derived logic" that might get weird to start breaking out of the settings class...at least as part of this task. It'll start requiring DI of different settings objects or adapters that consume a settings object to provide getters with a bit more logic to them. I think I see your point though and gravitating perhaps towards Single Responsibility Principle, with the Settings currently being a "bag of everything" and needing to trim it down to small bags of settings tailored toward different responsibilities. That would certainly break the interface down into more logical pieces, but perhaps part of another task to refactor.

As far as this function you were mentioning as an alternative:

```kotlin
override fun refreshPodcastsOnResume(context: Context): Boolean =
    settings.syncOnMeteredNetwork() || Network.isUnmeteredConnection(context)
```

My original implementation went this route, but `Network.isUnmeteredConnection(context)` becomes untestable because its type and would require Network to be injected. It can't be stubbed out to a mock from what I could tell, but I could be wrong. Injecting a POD for metered/unmetered made this testable and avoided making changes to the Network object.